flareon/src/db/migrations.rs

```diff
@@ -613,6 +648,52 @@ impl From<ColumnType> for sea_query::ColumnType {
     }
 }
 
+/// A migration dependency: a relationship between two migrations that tells the
+/// migration engine which migrations need to be applied before
+/// others.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct MigrationDependency {
+    inner: MigrationDependencyInner,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+enum MigrationDependencyInner {
```

Comment:
They are not, one is using `String` so that it can be dynamically created in the CLI, and the other has `&'static str`s which can work in `const` contexts. We have the same problem with `DynOperation` - perhaps we could use generics or macros to deduplicate this, for now I've created a task: https://github.com/orgs/flareon-rs/projects/2/views/7?pane=issue&itemId=88857597

Reply:
Makes sense!